x/gateway/keeper/msg_server_unstake_gateway_test.go

```diff
@@ -9,14 +9,83 @@ import (
 
 	keepertest "github.com/pokt-network/poktroll/testutil/keeper"
 	"github.com/pokt-network/poktroll/testutil/sample"
+	testsession "github.com/pokt-network/poktroll/testutil/session"
 	"github.com/pokt-network/poktroll/x/gateway/keeper"
 	"github.com/pokt-network/poktroll/x/gateway/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
 )
 
 func TestMsgServer_UnstakeGateway_Success(t *testing.T) {
```

Comment:
I'd love to see two more tests (use AI to help you):

- `func TestMsgServer_UnstakeGateway_DelegatingApplicationCheck_Success`
- `func TestMsgServer_UnstakeGatewayAndRestake_DelegatingApplicationCheck_Success`

Reply:
If I'm correctly understanding what the tests should do, those should be covered by:
[TestMsgServer_UndelegateFromGateway_UndelegateFromUnbondingGateway](https://github.com/pokt-network/poktroll/blob/13adb456f31cf7904741af5792b8a5c962f7b349/x/application/keeper/msg_server_undelegate_from_gateway_test.go#L525)

And

[TestMsgServer_UnstakeGateway_RestakeSuccess](https://github.com/pokt-network/poktroll/blob/13adb456f31cf7904741af5792b8a5c962f7b349/x/gateway/keeper/msg_server_unstake_gateway_test.go#L153)

Reply:
Will trust you on this one.